src/main/java/pl/javastart/streamsexercise/PaymentService.java

```diff
@@ -19,91 +26,132 @@ class PaymentService {
     Znajdź i zwróć płatności posortowane po dacie rosnąco
      */
     List<Payment> findPaymentsSortedByDateAsc() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> paymentsSortedByDateAsc = allPaymentsList();
+        paymentsSortedByDateAsc.sort(Comparator.comparing(Payment::getPaymentDate));
+        return paymentsSortedByDateAsc;
     }
 
     /*
     Znajdź i zwróć płatności posortowane po dacie malejąco
      */
     List<Payment> findPaymentsSortedByDateDesc() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> paymentsSortedByDateAsc = findPaymentsSortedByDateAsc();
+        paymentsSortedByDateAsc.sort(Comparator.comparing(Payment::getPaymentDate).reversed());
+        return paymentsSortedByDateAsc;
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów rosnąco
      */
+
     List<Payment> findPaymentsSortedByItemCountAsc() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> allPaymentsList = allPaymentsList();
+        allPaymentsList.sort(Comparator.comparing(payment -> payment.getPaymentItems().size()));
+        return allPaymentsList;
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów malejąco
      */
     List<Payment> findPaymentsSortedByItemCountDesc() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> paymentsSortedByItemCountAsc = findPaymentsSortedByItemCountAsc();
+        return paymentsSortedByItemCountAsc.stream().sorted((a, b) -> -1).toList();
     }
 
     /*
     Znajdź i zwróć płatności dla wskazanego miesiąca
      */
     List<Payment> findPaymentsForGivenMonth(YearMonth yearMonth) {
-        throw new RuntimeException("Not implemented");
+        List<Payment> allPaymentsList = allPaymentsList();
+        return allPaymentsList.stream().filter(payment -> payment.getPaymentDate().getMonthValue() == yearMonth.getMonthValue()).toList();
     }
 
     /*
     Znajdź i zwróć płatności dla aktualnego miesiąca
      */
     List<Payment> findPaymentsForCurrentMonth() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> allPaymentsList = allPaymentsList();
+        return allPaymentsList.stream()
+                .filter(payment -> payment.getPaymentDate().getMonthValue() == dateTimeProvider.yearMonthNow().getMonthValue()
+                        && payment.getPaymentDate().getYear() == dateTimeProvider.yearMonthNow().getYear()).toList();
```

Comment:
Jeżeli poprawisz metodę `findPaymentsForGivenMonth`, to będziesz mógł ją tutaj wykorzystać